app/models/main.py

```diff
@@ -635,6 +622,32 @@ def serialize_agreed_variation(agreed_variation, with_users=False):
             "agreedUserEmail": user.email_address,
         })
 
+    @property
+    def application_status(self):
```

Comment:
So perhaps the issue here is that making something a `property` can quite strongly suggest that it's quite "quick" to run (for some value of quick). But here we're running a DB query, so that isn't necessarily the case.

So maybe this is a nitpick but I think probably it shouldn't be a property. Not unless we can reasonably cache the value, anyway, which might add some unnecessary complexity.

Reply:
I suppose - although it basically acts the same as a lazy loaded relationship on the model, which we have in plenty of other places? All those would need to hit the DB to load the information they need.

Reply:
^this. Simply accessing a lazy loaded relationship (of which we have plenty) can cause signifacant database hits. Done as a hybrid property has the advantage that we can also filter by the property without having to pull extra things out of the db.